### PR TITLE
Replaced txt-based door code with DEFINE_FOR_X

### PR DESCRIPTION
**Pull request by KwinOriginal**
## About The Pull Request

As stated above, hope it's all good 

## Why It's Good For The Game

Easier to dive into the code, you can see what access doors have instead of looking for that strange number-to-mob table

## Changelog
:cl:
refactor: Replaced odd txt-based door code with DEFINE_FOR_X
/:cl:

### CODE REVIEW

File: code/__DEFINES/access.dm [hunk 1]
```diff
@@ -3,7 +3,7 @@
 It's best not to mess with the numbers of the regular access levels because
 most of them are tied into map-placed objects. This should be reworked in the future.*/
 //WE NEED TO REWORK THIS ONE DAY.  Access levels make me cry - Apophis
-#define ACCESS_MARINE_CAPTAIN 	1
```
Comment (spookydonut) [inline code/__DEFINES/access.dm]: dont make whitespace changes like this

**Comment by KwinOriginal** [comment]
Oof, I'll change it back soon, thanks

**Comment by Rohesie** [comment]
:+1:

File: code/game/machinery/doors/airlock_types.dm [hunk 2]
```diff
@@ -310,213 +310,173 @@
...
 /obj/machinery/door/airlock/almayer/marine/alpha/sl
 	name = "\improper Alpha Squad Leader Preparations"
-	req_access_txt ="12;15"
-	req_one_access_txt =  "0"
-	dir = 2
-	opacity = 0
-	glass = 1
+	req_access = list(ACCESS_MARINE_LEADER, ACCESS_MARINE_ALPHA)
+	req_one_access =  list()
+	dir = SOUTH
```
Comment (DominikPanic) [inline code/game/machinery/doors/airlock_types.dm]: Could make it even more compact by moving the dir to the parent on these

Comment (KwinOriginal) [reply code/game/machinery/doors/airlock_types.dm]: Right, I'll get on it quickly.

Comment (KwinOriginal) [reply code/game/machinery/doors/airlock_types.dm]: Done

**Comment by Menshin** [comment]
:+1:

**Comment by DominikPanic** [comment]
> 
> 
> You can probably just use null instead of list()

I'd argue list is better for clarity. This has caused issue before: #1247

**Comment by spookydonut** [comment]
> 
> 
> > You can probably just use null instead of list()
> 
> I'd argue list is better for clarity. This has caused issue before: #1247

using list() slows down init, and the access code deals with null req_access and req_one_access fine.

File: code/game/machinery/doors/airlock_types.dm [hunk 3]
```diff
@@ -557,8 +501,8 @@
 /obj/machinery/door/airlock/hatch/cockpit
 	icon = 'icons/obj/doors/almayer/dropship1_pilot.dmi'
 	name = "\improper Cockpit"
-	req_access_txt = "22"
-	req_one_access_txt = "0"
+	req_access = list(ACCESS_MARINE_DROPSHIP)
+	req_one_access = list()
```
Comment (spookydonut) [inline code/game/machinery/doors/airlock_types.dm]:
```suggestion
	req_one_access = null
```

Comment (KwinOriginal) [reply code/game/machinery/doors/airlock_types.dm]: I changed it for now. If you can agree on how you want it finally, let me know.

File: code/game/machinery/doors/airlock_types.dm [hunk 4]
```diff
@@ -259,21 +259,21 @@
...
 	icon = 'icons/obj/doors/almayer/securedoor.dmi'
-	req_access_txt = "19"
+	req_access = list(ACCESS_MARINE_BRIDGE)
 
 /obj/machinery/door/airlock/almayer/maint
 	name = "\improper Maintenance Hatch"
 	icon = 'icons/obj/doors/almayer/maintdoor.dmi'
-	req_access_txt = "0"
-	req_one_access_txt = "2;7"
+	req_access = null
```
Comment (Rohesie) [inline code/game/machinery/doors/airlock_types.dm]:
```suggestion
```
This should be the access by default, so no need to define it here.

File: code/game/machinery/doors/airlock_types.dm [hunk 5]
```diff
@@ -259,21 +259,21 @@
...
 /obj/machinery/door/airlock/almayer/engineering
 	name = "\improper Engineering Airlock"
 	icon = 'icons/obj/doors/almayer/engidoor.dmi'
-	opacity = 0
-	glass = 1
-	req_access_txt = "0"
-	req_one_access_txt = "2;7"
+	opacity = FALSE
+	glass = TRUE
+	req_access = null
```
Comment (Rohesie) [inline code/game/machinery/doors/airlock_types.dm]:
```suggestion
```

File: code/game/machinery/doors/airlock_types.dm [hunk 6]
```diff
@@ -282,16 +282,16 @@
 /obj/machinery/door/airlock/almayer/medical
 	name = "\improper Medical Airlock"
 	icon = 'icons/obj/doors/almayer/medidoor.dmi'
-	req_access_txt = "0"
-	req_one_access_txt =  "2;8;19"
+	req_access = null
```
Comment (Rohesie) [inline code/game/machinery/doors/airlock_types.dm]:
```suggestion
```

File: code/game/machinery/doors/airlock_types.dm [hunk 7]
```diff
@@ -282,16 +282,16 @@
...
 /obj/machinery/door/airlock/almayer/medical/glass
 	name = "\improper Medical Airlock"
 	icon = 'icons/obj/doors/almayer/medidoor_glass.dmi'
-	opacity = 0
-	glass = 1
-	req_access_txt = "0"
-	req_one_access_txt =  "2;8;19"
+	opacity = FALSE
+	glass = TRUE
+	req_access = null
```
Comment (Rohesie) [inline code/game/machinery/doors/airlock_types.dm]:
```suggestion
```

File: code/game/machinery/doors/airlock_types.dm [hunk 8]
```diff
@@ -310,213 +310,157 @@
...
+	glass = TRUE
 
 /obj/machinery/door/airlock/almayer/marine/requisitions
 	name = "\improper Requisitions Bay"
 	icon = 'icons/obj/doors/almayer/prepdoor.dmi'
-	req_access_txt = "0"
-	req_one_access_txt =  "2;21"
-	opacity = 0
-	glass = 1
+	req_access = null
```
Comment (Rohesie) [inline code/game/machinery/doors/airlock_types.dm]:
```suggestion
```

File: code/game/machinery/doors/airlock_types.dm [hunk 9]
```diff
@@ -310,213 +310,157 @@
...
 	icon = 'icons/obj/doors/almayer/prepdoor_alpha.dmi'
-	req_access_txt = "9"
-	req_one_access_txt =  "2;15"
-	opacity = 0
-	glass = 1
+	req_access = list(ACCESS_MARINE_PREP)
+	req_one_access = list(ACCESS_MARINE_LOGISTICS, ACCESS_MARINE_ALPHA)
+	opacity = FALSE
+	glass = TRUE
+	dir = SOUTH
```
Comment (Rohesie) [inline code/game/machinery/doors/airlock_types.dm]:
```suggestion
```
Direction is something that's usually var-edited in maps, no?

File: code/game/machinery/doors/airlock_types.dm [hunk 10]
```diff
@@ -310,213 +310,157 @@
...
 	icon = 'icons/obj/doors/almayer/prepdoor_bravo.dmi'
-	req_access_txt = "9"
-	req_one_access_txt =  "2;16"
-	opacity = 0
-	glass = 1
+	req_access = list(ACCESS_MARINE_PREP)
+	req_one_access = list(ACCESS_MARINE_LOGISTICS,ACCESS_MARINE_BRAVO)
+	opacity = FALSE
+	glass = TRUE
+	dir = SOUTH
```
Comment (Rohesie) [inline code/game/machinery/doors/airlock_types.dm]:
```suggestion
```

File: code/game/machinery/doors/airlock_types.dm [hunk 11]
```diff
@@ -310,213 +310,157 @@
...
 	icon = 'icons/obj/doors/almayer/prepdoor_charlie.dmi'
-	req_access_txt = "9"
-	req_one_access_txt =  "2;17"
-	opacity = 0
-	glass = 1
+	req_access = list(ACCESS_MARINE_PREP)
+	req_one_access = list(ACCESS_MARINE_LOGISTICS, ACCESS_MARINE_CHARLIE)
+	opacity = FALSE
+	glass = TRUE
+	dir = SOUTH
```
Comment (Rohesie) [inline code/game/machinery/doors/airlock_types.dm]:
```suggestion
```

File: code/game/machinery/doors/airlock_types.dm [hunk 12]
```diff
@@ -557,8 +501,8 @@
 /obj/machinery/door/airlock/hatch/cockpit
 	icon = 'icons/obj/doors/almayer/dropship1_pilot.dmi'
 	name = "\improper Cockpit"
-	req_access_txt = "22"
-	req_one_access_txt = "0"
+	req_access = list(ACCESS_MARINE_DROPSHIP)
+	req_one_access = null
```
Comment (Rohesie) [inline code/game/machinery/doors/airlock_types.dm]:
```suggestion
```

**Pull request merged**